Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp

```diff
@@ -244,14 +244,29 @@ static String originFilePath(const String& directory)
     return FileSystem::pathByAppendingComponent(directory, OriginStorageManager::originFileIdentifier());
 }
 
+void NetworkStorageManager::writeOriginToFileIf(const WebCore::ClientOrigin& origin, const Function<bool()>& condition)
```

Comment:
Why aren't we doing the condition check at call sites?

Reply:
Because the first checks in the functions are cheaper; if we can return early here, we don't need to call condition() which may involve file operation.